### PR TITLE
Support for Item categorization in Jenkins 2

### PR DESCRIPTION
**Pull request by recena**
This PR belongs to this [JENKINS-31162](https://issues.jenkins-ci.org/browse/JENKINS-31162) issue and allows to categorize the Maven projects in a right way.

@reviewbybees /cc @daniel-beck

### CODE REVIEW

File: src/main/java/hudson/maven/MavenModuleSet.java [hunk 1]
```diff
@@ -1365,6 +1365,39 @@ public String getDisplayName() {
...
+         * TODO: Override when the baseline is upgraded to 2.x
+         *
+         * @return A string it represents a ItemCategory identifier.
+         */
+        public String getCategoryId() {
+            return "itemcategory-standaloneprojects";
+        }
+
+        /**
+         * Needed if it wants Maven projects are categorized in Jenkin 2.x.
```
Comment (oleg-nenashev) [inline src/main/java/hudson/maven/MavenModuleSet.java]: "Jenkin"

File: src/main/java/hudson/maven/MavenModuleSet.java [hunk 2]
```diff
@@ -1365,6 +1365,39 @@ public String getDisplayName() {
             return Messages.MavenModuleSet_DiplayName();
         }
 
+        /**
+         * Needed if it wants Maven projects are categorized in Jenkin 2.x.
```
Comment (oleg-nenashev) [inline src/main/java/hudson/maven/MavenModuleSet.java]: same

File: src/main/java/hudson/maven/MavenModuleSet.java [hunk 3]
```diff
@@ -1365,6 +1365,39 @@ public String getDisplayName() {
...
+        }
+
+        /**
+         * Needed if it wants Maven projects are categorized in Jenkin 2.x.
+         *
+         * TODO: Override when the baseline is upgraded to 2.x
+         *
+         * @return A string with the Item description.
+         */
+        public String getDescription() {
```
Comment (oleg-nenashev) [inline src/main/java/hudson/maven/MavenModuleSet.java]: May be null, right? Makes sense to process or annotate it somehow

**Comment by oleg-nenashev** [comment]
I don't understand the purpose of mass property file modifications excepting the old key deletion. Should we review it? Or should it be reverted?

File: src/main/java/hudson/maven/MavenModuleSet.java (hunk 3 above)
Comment (recena) [inline src/main/java/hudson/maven/MavenModuleSet.java]: @oleg-nenashev It seems this plugins is not using this kind of annotations. Could @aheritier confirm it?

**Comment by recena** [comment]
@oleg-nenashev My IDE used uppercase for the character codes but the real change is:

- [x] `newJobDetail*.properties` need no longer
- [x] Its descriptions are been moved to `Messages*.properties`

The last one was very humdrum, specially because I'm doing that in other plugins involve in JENKINS-31162.

File: src/main/java/hudson/maven/MavenModuleSet.java (hunk 3 above)
Comment (aheritier) [inline src/main/java/hudson/maven/MavenModuleSet.java]: Yes @recena it was never updated. To be done (perhaps) one day but in another PR I think ..

Comment (recena) [inline src/main/java/hudson/maven/MavenModuleSet.java]: @aheritier I can send another one in other to use it.

**Comment by aheritier** [comment]
@recena For now :bug: excepted if you prove that the CI failure doesn't come from your changes (I didn't verified)
Otherwise I'm :+1: for the proposed changes (even if the case change creates many noise in the commit, it isn't critical, at least it is doing some cleanup/unification)
Myself I love clean history and to rewrite it in PRs thusfrom my POV 4e4aa233b1c9ae525086face6c89817caae57676 should be squashed into d7e20b0d6f43b5f075c7f45a16fd9718b34b84be ( 4e4aa233b1c9ae525086face6c89817caae57676 has really no value in the project history by itself)

**Comment by jglick** [comment]
@aheritier you need to review my #60, and that has to merged and `master` merged into this one, before this PR can even be evaluated so far as tests are concerned.

**Comment by recena** [comment]
@aheritier squashed ;)

File: src/main/resources/hudson/maven/Messages_da.properties [hunk 4]
```diff
@@ -24,21 +24,24 @@ MavenModuleSetBuild.DiscoveredModule=Opdagede et nyt modul {0} {1}
 MavenBuild.Triggering=Starter et nyt byg af {0}
 MavenBuild.KeptBecauseOfParent=Beholdt da {0} er gemt
 MavenModuleSetBuild.FailedToParsePom=Kunne ikke parse POM''erne
-MavenBuilder.Waiting=Venter p\u00e5 at Jenkins bliver f\u00e6rdig med at samle data
+MavenBuilder.Waiting=Venter p\u00E5 at Jenkins bliver f\u00E6rdig med at samle data
```
Comment (jglick) [inline src/main/resources/hudson/maven/Messages_da.properties]: @recena this is unreviewable. You can use [this script](https://github.com/jenkinsci/jenkins/blob/27aeca9ee28999e5532b6e2a285e9c3306a4ac16/core/move-l10n.groovy) with modifications to the basedir for moving I18N lines without mangling unrelated parts of the text.

**Comment by jglick** [comment]
If this is downstream of some core PR you need to link to it. https://github.com/jenkinsci/jenkins/pull/2117 I presume?

**Comment by jglick** [comment]
@recena merge with `master` to fix tests.

**Comment by recena** [comment]
@jglick Technically it is not a downstream, this PR can be merged separately.

File: src/main/resources/hudson/maven/Messages_da.properties (hunk 4 above)
Comment (recena) [inline src/main/resources/hudson/maven/Messages_da.properties]: @jglick Thanks. In this case the only different is the uppercase.

**Comment by jglick** [comment]
> this PR can be merged separately

Sure, but the purpose is to work with that PR. For reviewers it is helpful to link directly to the core PR from your plugin PR’s description, so the change can be seen in context.

File: src/main/resources/hudson/maven/Messages_da.properties (hunk 4 above)
Comment (jglick) [inline src/main/resources/hudson/maven/Messages_da.properties]: Right, but it is hardly possible to _see_ that from looking at the diff.

**Comment by jglick** [comment]
> it is helpful to link directly to the core PR from your plugin PR’s description

It would also be helpful for the core PR’s summary to include a list of related plugin PRs.

**Comment by recena** [comment]
@jglick 
> If this is downstream of some core PR you need to link to it. jenkinsci/jenkins#2117 I presume?

You know perfectly that this PR is related to https://github.com/jenkinsci/jenkins/pull/2117

**Comment by recena** [comment]
@jglick 

> It would also be helpful for the core PR’s summary to include a list of related plugin PRs.

I've just added the two PR are on the way. I'll keep the list updated.

**Comment by recena** [comment]
@aheritier After merge from `master` (thanks to @jglick) the tests are running fine.

**Comment by aheritier** [comment]
ok thus perfect @recena 
:bee:

**Comment by jglick** [comment]
:bee:

**Comment by recena** [comment]
@reviewbybees done

**Pull request merged**